jbi/actions/default.py

```diff
@@ -96,8 +98,23 @@ def __call__(self, context: ActionContext) -> ActionResult:
 
         responses = tuple()  # type: ignore
 
+        incomplete = False
+
         for step in self.steps[context.operation]:
-            context, step_responses = step(context=context, **self.parameters)
+            try:
+                context, step_responses = step(context=context, **self.parameters)
+            except Exception:
+                if len(responses) > 0:
+                    # Count the number of workflows that produced at least request,
+                    # but could not complete entirely successfully.
+                    statsd.incr("jbi.bugzilla.aborted.count")
+                raise
+
+            if not step_responses:
+                # If the step did not generate HTTP response, it means the step
+                # skipped its operation (for various reasons, safety checks, etc.)
+                incomplete = True
```

Comment:
Not generating an HTTP response might not always be a signal that the step didn't complete as expected. Take:

```python
def update_issue_summary(context: ActionContext, **parameters):
    """Update the Jira issue's summary if the linked bug is modified."""

    bug = context.bug
    issue_key = context.jira.issue

    if "summary" not in context.event.changed_fields():
        return context, ()
```

Us returning a context object early with no response isn't a signal that something went wrong -- it just means the summary wasn't updated, so we don't need to do anything to the linked issue. In this case, the step function behaved 100% correctly.